core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala

```diff
@@ -352,6 +560,56 @@ case class MultiClassificationMetricsTopK
   Error: Seq[Double]
 ) extends EvaluationMetrics
 
+/**
+ * Metrics for multi-classification confusion matrix. It captures the confusion matrix of records:
+ * (1) of which the labels fall into the set of the topK most frequent labels
+ * (2) of which the probability of the true label exceeds the corresponding threshold
+ *
+ * @param topK cutoff value for the topK most frequent labels
+ * @param thresholds thresholds for predicted probabilities. Values must be between 0.0 and 1.0
+ * @param confusionMatrixTopKByThreshold a 3D array that stores the counts. 1st dim is thresholds, 2nd dim is labels
+ *                                       and 3rd dim is predictions
+ */
+case class MulticlassConfusionMatrixTopKByThreshold
+(
+  topK: Int,
+  thresholds: Seq[Double],
+  confusionMatrixTopKByThreshold: Seq[Seq[Seq[Long]]]
```

Comment:
I think we should try and keep this close to the way Spark treats the multiclass confusion matrices, if possible. They use their `Matrix` type for the actual confusion matrix, see eg. https://github.com/apache/spark/blob/v2.4.0/mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala#L67. I'd suggest a `Seq[Matrix]` type for the confusion matrices by threshold.

Reply:
I tried returning a Matrix type as the confusion matrix, but it's causing Json deserialization error in both `MultiClassificationModelSelectorTest` and `ModelSelectorSummaryTest` (error msg: Caused by: com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of org.apache.spark.mllib.linalg.Matrix, problem: abstract types either need to be mapped to concrete types, have custom deserializer, or be instantiated with additional type information). Adding a Json serialization annotation doesn't help. I think it needs to be a concrete type in order to be extracted correctly.

In the updated PR, I keep the returned type as an Array. @Jauntbox